### DISCUSSION TITLE
SEO optimization

### DISCUSSION THREAD

**Discussion by rugginic** [chipzoller/hugo-clarity#139] [Q&A]
Hi all,
I have been writing some new articles lately and it seems that Google doesn't like my site. The articles wouldn't appear on Google search even if I type the title exactly as it is, not even in 3rd or 4th page of results. Now, I'm not a SEO expert. I did try Google dev tool Lighthouse and my pages hardly score 70/100 for SEO. I wonder if someone here could help me out with some good advice to ramp up on search engines.
Thank you,
Nicola

**Comment by onweru** [comment]
@rugginic, as a start, have you submitted your site via https://search.google.com/search-console/about?

**Reply by rugginic** [reply]
hi @onweru!
Thanks for that! I had not, but I did now!
When I check for SEO with Lighthouse I get the following 2 problems:
`Links do not have descriptive text 11 links found`
`Descriptive link text helps search engines understand your content`
And this is for all the links to my posts. All my posts have the description in the front matter, so I'm not sure what this message is complaining about and how can I fix it.
**- How can I add a descriptive text to my links?**

The second problem is:
`Image elements do not have [alt] attributes`
And it refers to the thumbnail image in the post's front matter, the website logo and the "apple-touch-icon"
**- How can I add the alt attribute to the thumbnail image?**
**- How can I add the alt attribute to the blog main logo (from config.toml) and the "apple-touch-icon"?**

Thanks in advance,
Nicola

**Reply by onweru** [reply]
@rugginic, you're welcome. Can you share your sites url? I will look into it and advise accordingly.

**Reply by rugginic** [reply]
@onweru Sure! Awesome Thanks!
My site: https://itdeepdive.com/

**Reply by onweru** [reply]
@rugginic, this update fixes 3b52adf the lighthouse issues you highlighted.

**Reply by rugginic** [reply]
Hi @onweru Thanks again for your priceless efforts!!
I've download the latest commit.
Apparently you fixed the thumbnail images, but the link description, logo and apple-touch-icon are still affected:
![image](https://user-images.githubusercontent.com/68188977/107801440-fa120780-6d57-11eb-89bd-2f64d9fbb1f7.png)
![image](https://user-images.githubusercontent.com/68188977/107801548-1f9f1100-6d58-11eb-8786-53748dee0e22.png)
![image](https://user-images.githubusercontent.com/68188977/107801679-4a896500-6d58-11eb-9318-38fea6f4490b.png)

Thanks,
Nicola

**Reply by onweru** [reply]
You're most welcome; I will fix those as well.

**Reply by onweru** [reply]
Nicola, please check if I nailed all the outstanding ones. Now, within the markdown files for each post, you would need to author link in the following format

```markdown
[Link Label](http://www.google.com/ "Google")
```

`Google` would be the value of the `title` attribute for the links.

**Reply by rugginic** [reply]
Hi @onweru!
Definitely resolved the Alt description for the logo and apple-touch-icon!!  The score increased but apparently links still don't have descriptive text:
![image](https://user-images.githubusercontent.com/68188977/107861807-315be380-6e40-11eb-88e6-f5787e2f4f64.png)

Sorry is not clear for me what you mean with: "_within the markdown files for each post, you would need to author link in the following format"_
These links are automatically generated when I create a new post "hugo new post/bla-bla-bla.md" So I'm not sure where and how I can add that Alt text to my links. Is it in the from matter of the post? or just anywhere in the page? Or in config.toml?

Thanks,
Nicola

**Reply by rugginic** [reply]
Hi @onweru I think I've found the problem...
![image](https://user-images.githubusercontent.com/68188977/107863980-15f8d480-6e50-11eb-94ce-cc3f8f5088c5.png)

The button READ MORE links to the page "https://site.com/post/2020-new-post/" but the link text is "READ MORE". I think it should be the page title. same as here:
`<a href="https://itdeepdive.com/2021/01/my-cka-exam-experience/" title="My CKA Exam Experience">My CKA Exam Experience</a>`

**Reply by abit82** [reply]
First, your pages are indexed (you can check it by using the "site:" search parameter in Google search). Your page has one ranking "vmware workstation player evaluation" on position 97 - so basically no ranking. Even 100 % SEO Score doesn't mean you will rank now. 

Read Now as anchor everywhere is confusing of course, you say in an anchor what the page is about (so the title is not bad). But it is not a big problem, if other links with keywords exists.

Images without alt - if the image is useful for the content (not used for design only) it should have an alt-attribute. But that's the Basics, more web development than SEO.
